packages/client-sdk-nodejs/src/errors/cache-service-error-mapper.ts

```diff
@@ -76,8 +80,19 @@ export class CacheServiceErrorMapper
         return new UnknownServiceError(...errParams);
       case Status.UNAVAILABLE:
         return new ServerUnavailableError(...errParams);
-      case Status.NOT_FOUND:
-        return new NotFoundError(...errParams);
+      case Status.NOT_FOUND: {
```

Comment:
this is a key thing in my changes - after playing with the API and discussion with some other folks we are feeling like we don't want users to have to deal with the full error handling patterns in order to tell when an item just did not exist. But the server is returning it as an error. So this code is what we need in order to disambiguate between different types of NOT FOUND.